src/components/modules/TagPostsSingleColumn/posts-helpers.tsx

```diff
@@ -23,12 +23,16 @@ const POST_COMPONENT_ATTRS: PostComponentAttrMap = {
   MD: { layoutVariant: 'medium' }
 };
 
-const ORDERS = [['LG', 'MD', 'MD', 'MD', 'AD'], ['MD', 'MD', 'AD', 'MD', 'LG']];
+const ORDERS = [
+  ['LG', 'MD', 'MD', 'MD', 'AD'], 
+  ['MD', 'MD', 'AD', 'MD', 'LG'], 
+  ['MD', 'MD', 'MD', 'AD']
+];
 
 export function mapPostsToGrid(order: GridOrder, posts: PostType[]) {
   const orderSpec = ORDERS[order - 1];
   const postQueue = posts.slice(0).reverse();
-  if (postQueue.length != GRID_POST_LENGTH) {
+  if (postQueue.length != GRID_POST_LENGTH && orderSpec.length === 5) {
```

Comment:
I think a better solution to this is to get rid of GRID_POST_LENGTH, and determine it dynamically based on orderSpec, something like:
const orderSpecPostCount = orderSpec.filter(type => type == 'LG' || type== 'MD').length;
if (postQueue.length != orderSpecPostCount) ...

Reply:
Revised posts-helpers.tsx per your suggestion.